addons/l10n_eg_edi_eta/models/account_move.py

```diff
@@ -0,0 +1,186 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import base64
+import hashlib
+from datetime import datetime
+
+import pytz
+from asn1crypto import cms, x509, algos, core, tsp
```

Comment:
`asn1crypto` is a new "external dependency" that needs to be declared in the `__manifest__`, it can't be added as a new required dependency in a stable series.

Besides, it would perhaps be better to move this logic on the `hw` module, which already requires special installation. The `cms` stuff doesn't seem to be used for anything else than preparing the digital signature. This way the main server won't need any extra libraries than current stable versions, making deployment easier.

Reply:
But the opposite is also true: if you put the logic on the server and there is a bug, you can fix it there for all, instead of having to meddle with local installations.  (it was like that before)

Reply:
> But the opposite is also true: if you put the logic on the server and there is a bug, you can fix it there for all, instead of having to meddle with local installations. (it was like that before)

Yes, it's a trade-off, which could be decided based on the expected number of patches in that code, and the expected number of customers where the signature code will have to be deployed.

Either way, this is not a security issue, it's a deployment concern, so feel free to choose either option in the end, as long as those new dependencies are declared as external (optional) deps.